code/modules/virology/virology_goals.dm

```diff
@@ -0,0 +1,196 @@
+GLOBAL_LIST_INIT(virology_goals, list(new /datum/virology_goal/property_symptom, new /datum/virology_goal/virus, new /datum/virology_goal/virus/stealth))
+GLOBAL_LIST_EMPTY(archived_virology_goals)
+
+/datum/virology_goal
+	var/name = "Generic Virology Goal"
+	/// The amount of units currently already delivered  
+	var/delivered_amount = 0  
+	/// The amount of units of the required virus that must be delivered for the completion of this goal  
+	var/delivery_goal = 15
+	var/completed = FALSE
+
+/datum/virology_goal/proc/get_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/get_ui_report()
+	return "Complete this goal."
+
+/datum/virology_goal/proc/check_completion(list/reagent_list)
+	check_total_virology_goals_completion()
+	return TRUE
+
+/datum/virology_goal/proc/check_for_duplicate()
+	return TRUE
+
+/proc/check_total_virology_goals_completion()
+	for(var/datum/virology_goal/V as anything in GLOB.virology_goals)
+		if(!V.completed)
+			return
+	GLOB.archived_virology_goals += GLOB.virology_goals
```

Comment:
I don't remember if I asked this before, but what if we also removed it from the normal goals here? That way we also don't have to check if there are completed goals in the list in other places

Reply:
Sorry but iam not sure what exactly you mean by "it"

Reply:
Looking back, I am an idiot
You overwrite the `GLOB.virology_goals` completely after this, mb whoops

Reply:
happens